source/bootloader_common.h

```diff
@@ -80,40 +82,36 @@ void print_progress_function(uint32_t progress, uint32_t total);
 #undef tr_debug
 #endif
 #if 0
-#define tr_debug(fmt, ...)   printf("[DBG ] " fmt "\r\n", ##__VA_ARGS__)
+#define tr_debug(fmt, ...)   boot_debug("[DBG ] " fmt "\r\n")
 #else
 #define tr_debug(...)
 #endif
 
 #ifdef tr_info
 #undef tr_info
 #endif
-#define tr_info(fmt, ...)    printf("[BOOT] " fmt "\r\n", ##__VA_ARGS__)
+#define tr_info(fmt, ...)    boot_debug("[INF ] " fmt "\r\n")
 
 #ifdef tr_warning
 #undef tr_warning
 #endif
-#define tr_warning(fmt, ...) printf("[WARN] " fmt "\r\n", ##__VA_ARGS__)
+#define tr_warning(fmt, ...) boot_debug("[WARN ] " fmt "\r\n")
```

Comment:
I guess having extra space for DBG,INF and ERR was to have constant width with all different types-> not needed with "WARN" is it fills the space on it's own.

Reply:
Done